docs/GroupDefs.hpp

```diff
@@ -487,6 +487,12 @@
  * \brief Functions and classes for manipulating HDF5 files
  */
 
+/*!
+ * \defgroup InitializationGroup Initialization
+ * \brief Actions and metafunctions used for initialization of parallel
+ * components.
```

Comment:
I think we should defer that to later since then there will be more tests that we can get snippets from. I think it should also be in `docs/DevGuide` so as to avoid making the `GroupDefs.hpp` too long (I know I've written docs in here in the past, and now regret it. Some have been moved).